lib/diagnostic.rb

```diff
@@ -27,7 +27,22 @@ def example
 # responds successfully and lists all examples.
 
 RSpec.describe 'Examples API' do
-  # your test(s) here
+  it 'lists all examples' do
+      # get is a functions -- takes a string, makes a get request to that
+      # address
+      get '/examples'
+      # expect a 20x response
+      expect(response).to be_success
+
+      # sets variable articles_response to the body of the http response
+      examples_response = JSON.parse(response.body)
+      # expect list of articles from server to be same length as list of
+      # articles we created
+      expect(examples_response.length).to eq(examples.count)
+      # expect first articles title to be the same as the articles title we put
+      # in
+      expect(examples_response.first['title']).to eq(example['title'])
+    end
 end
```

Comment:
I'll give you this one, but note that you did not define your variables `examples` and `example`.

```rb
RSpec.describe 'Examples API' do
  def examples
    Example.all
  end

  def example
    Example.first
  end

  it 'lists all examples' do
    get '/examples'

    expect(response).to be_success

    examples_response = JSON.parse(response.body)
    expect(examples_response.length).to eq(examples.count)
    expect(examples_response['examples'].first['title']).to eq(example.title)
  end
end
```